src/structs.jl

```diff
@@ -229,6 +236,29 @@ read(::ArrayType, buf, pos, len, b, ::Type{Tuple}, ::Type{eT}; kw...) where {eT}
     invalid(error, buf, pos, T)
 end
 
+@inline readjsonlines(buf, pos, len, b, ::Type{T}; kw...) where {T} = readjsonlines(buf, pos, len, b, T, Base.IteratorEltype(T) == Base.HasEltype() ? eltype(T) : Any; kw...)
+@inline readjsonlines(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT} = readjsonlinesarray(buf, pos, len, b, T, eT; kw...)
+
+@inline function readjsonlinesarray(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT}
+    vals = Vector{eT}(undef, 0)
+    if pos > len
+        @goto done
+    end
+    while true
+        # positioned at start of value
+        pos, y = read(StructType(eT), buf, pos, len, b, eT; kw...)
+        push!(vals, y)
+        pos += 1
```

Comment:
I think I'd like to be more consistent here. In read.jl, we do explicit checks for `\r`, `\n`, or `\r\n` as three supported newline possibilities, instead of just incrementing `pos`. See the code starting [here](https://github.com/quinnj/JSON3.jl/blob/28a3a8b0f15ffa2821d0f208a3ceec2bd6638bbb/src/read.jl#L374)

Reply:
I think the newlines are already taken care of in the call to `read` in line 249.  After that, `b` contains the first character of the next line (discovered by stepping through the code with a debugger).  However, `pos` was still pointing at the same location.

Reply:
Hmm, that sounds fishy to me; the code [here](https://github.com/quinnj/JSON3.jl/blob/28a3a8b0f15ffa2821d0f208a3ceec2bd6638bbb/src/structs.jl#L463) is where we would "finish" reading a struct and it immediately returns without consuming any more whitespace. It also seems fishy here that you're not updating `b` before the loop starts over and you call `read` again, like we do [here](https://github.com/quinnj/JSON3.jl/blob/28a3a8b0f15ffa2821d0f208a3ceec2bd6638bbb/src/structs.jl#L461).

Reply:
Actually, I think the whitespace is consumed in the [line above that](https://github.com/quinnj/JSON3.jl/blob/28a3a8b0f15ffa2821d0f208a3ceec2bd6638bbb/src/structs.jl#L462-L463).

And while I agree that it makes sense that `b` should be updated here, I can assert that the current code is working fine to read in large jsonl files (but it's not entirely clear why).

I'll dig into it a little more as I have time.

Reply:
Okay, I'm understanding a little more.  `b` definitely should be updated here--the reason it was "working" for me was because the character to be read was the start character (and my jsonl files don't have extra whitespace).

Reply:
I've updated the code.  The current `@wh` macro doesn't really work because whitespace works differently in jsonlines, so I created a separate `@jlwh` (json lines white space) macro.

There are still some corner cases that aren't going to work correctly here (e.g., whitespace at the end of the line, CR-LF (`\r\n`)), but this is what I have time for right this minute.  I can try to take a further look later or tomorrow.

Reply:
Okay, added tests for and handled all whitespace issues that I could think of.